atc/exec/check_step.go

```diff
@@ -204,11 +212,11 @@ func (step *CheckStep) run(ctx context.Context, state RunState, delegate CheckDe
 			state.StoreResult(step.planID, result.Versions[len(result.Versions)-1])
 		}
 
-		_, err = scope.UpdateLastCheckEndTime(true)
+		_, err = delegate.UpdateScopeLastCheckEndTime(scope, true)
 		if err != nil {
 			return false, fmt.Errorf("update check end time: %w", err)
 		}
-	} else {
+	} else if !step.plan.IsPeriodic() || delegate.IsManuallyTriggered() {
```

Comment:
What is this condition for?

Reply:
I was thinking Lidar check may not need to set run state, so that it save a `scope.LastestVersion()` call for each not-run checks. But maybe not a big deal. I can revert this also.